src/targets/services/helpers/model.js

```diff
@@ -189,12 +181,10 @@ export class Model {
    */
   getShares(nbShares) {
     // Initialize shares array
-    let shares = Array(nbShares)
-      .fill()
-      .map(() => ({
-        occurences: this.occurences.map(e => e.map(f => f)),
-        contributions: this.contributions
-      }))
+    let shares = Array(nbShares).fill({
+      occurences: this.occurences.map(e => e.map(f => f)),
```

Comment:
I might miss something, but why not just: 
```
let shares = Array(nbShares).fill({
      occurences: this.occurences,
      contributions: this.contributions
 })
 ```
 I don't get the double map, which is costful as it makes a copy